src/pipelines/token_classification.rs

```diff
@@ -0,0 +1,518 @@
+        }
+    }
+
+    /// Interface method to forward_t() of the particular models.
+    pub fn forward_t(&self,
+                     input_ids: Option<Tensor>,
+                     mask: Option<Tensor>,
+                     token_type_ids: Option<Tensor>,
+                     position_ids: Option<Tensor>,
+                     input_embeds: Option<Tensor>,
+                     train: bool) -> (Tensor, Option<Vec<Tensor>>, Option<Vec<Tensor>>)  {
+        match *self {
+            Self::Bert(ref model) => model.forward_t(input_ids,mask,token_type_ids,position_ids,input_embeds,train),
+            Self::DistilBert(ref model) => model.forward_t(input_ids,mask,input_embeds,train).expect("Error in distilbert forward_t"),
+            Self::Roberta(ref model) => model.forward_t(input_ids,mask,token_type_ids,position_ids,input_embeds,train),
+        }
+
+    }
+
+}
+
+/// # Holds the label mapping for the model, supports different model types.
+pub enum LabelMapping {
+    Bert(HashMap<i64, String>),
+    DistilBert(HashMap<i32, String>),
+}
+
+impl LabelMapping {
+    /// Interface method to HashMap.get()
+    fn get(&self, label: &i64) -> Option<&String> {
+        match *self {
+            Self::Bert(ref map) => map.get(label),
+            Self::DistilBert(ref map) => map.get(&(*label as i32)),
+        }
+    }
+}
+
+
+/// # TokenClassificationModel for Named Entity Recognition or Part-of-Speech tagging
+pub struct TokenClassificationModel {
+    tokenizer: TokenizerOption,
+    token_sequence_classifier: TokenClassificationOption, //e.g. BertForTokenClassification,
+    label_mapping: LabelMapping,
+    var_store: VarStore,
+}
+
+impl TokenClassificationModel {
+    /// Build a new `TokenClassificationModel`
+    ///
+    /// # Arguments
+    ///
+    /// * `config` - `TokenClassificationConfig` object containing the resource references (model, vocabulary, configuration) and device placement (CPU/GPU)
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    ///# fn main() -> failure::Fallible<()> {
+    /// use rust_bert::pipelines::token_classification::TokenClassificationModel;
+    ///
+    /// let model = TokenClassificationModel::new(Default::default())?;
+    ///# Ok(())
+    ///# }
+    /// ```
+    ///
+    pub fn new(config: TokenClassificationConfig) -> failure::Fallible<TokenClassificationModel> {
+        let config_path = download_resource(&config.config_resource)?;
+        let vocab_path = download_resource(&config.vocab_resource)?;
+        let weights_path = download_resource(&config.model_resource)?;
+        let merges_path = if let Some(merges_resource) = &config.merges_resource {
+            Some(download_resource(merges_resource).expect("Failure downloading resource"))
+        } else {
+            None
+        };
+        let device = config.device;
+
+        let tokenizer = TokenizerOption::from_file(config.model_type, vocab_path.to_str().unwrap(), merges_path.map(|path| path.to_str().unwrap()), false);
```

Comment:
This will default tokenizers to always lower_case. Can we add a the option to `TokenClassificationConfig` and pass it on to the tokenizer at this stage?

Reply:
Agreed, we don't want to force that indeed.